<!DOCTYPE html>
<html lang="en">
<head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <title>Data Science Portfolio</title>
      <style>
        * {
          box-sizing: border-box;
        }
        
        body {
          font-family: Arial, Helvetica, sans-serif;
        }
        
        /* Style the header */
        header {
          background-color: #666;
          padding: 30px;
          text-align: center;
          font-size: 35px;
          color: white;
        }
        
        /* Create two columns/boxes that floats next to each other */
        nav {
          float: left;
          width: 20%;
          height: 300px; /* only for demonstration, should be removed */
          background: #ccc;
          padding: 20px;
        }
        
        /* Style the list inside the menu */
        nav ul {
          list-style-type: none;
          padding: 0;
        }
        
        article {
          float: left;
          padding: 20px;
          width: 70%;
          background-color: #f1f1f1;
          height: 300px; /* only for demonstration, should be removed */
        }

        aside {
          float: left;
          width: 10%;
          background-color: aqua; 
        }
        
        /* Clear floats after the columns */
        section::after {
          content: "";
          display: table;
          clear: both;
        }
        
        /* Style the footer */
        footer {
          background-color: #777;
          padding: 10px;
          text-align: center;
          color: white;
        }
        
        /* Responsive layout - makes the two columns/boxes stack on top of each other instead of next to each other, on small screens */
        @media (max-width: 600px) {
          nav, article, aside {
            width: 100%;
            height: auto;
          }
        }
        </style>
</head>

<body>

<header>
  <h2>Data Science Portfolio</h2>
</header>

<section>
  <nav>
    <!-- <ul>
      <li>Web Development Project
        <ul>
          <li>WEGAN: Web-based Analytics Tool for Ecology Data</li>
        </ul>
  </li>
  <li>Bioinformatics Project
    <ul>
      <li>Comparative Genomics Analysis</li>
      <li>Strain-specific Primer Design</li>
      <li>Gene Variation Identification</li>
    </ul>
  </li>
  <li>Microbiome Epidemiology Project
    <ul>
      <li>Multivariate Regression between vitamin D use and infant gut microbiome</li>
      <li>Confounding Variables Identification using Directed Acyclic Graph</li>
      <li>Prenatal Factors and Bifidobacteria Abundance: Regression Analysis in qPCR and Sequencing Data</li>
      <li>Benchmarking algorithms on zero-excessive metagenome data</li>
      <li>Correlation network analysis of gut microbes using SparCC</li>
      <li>Proportionality of gut microbiome to mitigrate spurious correlation</li>
    </ul>
  </li>
  <li>Machine Learning Project
    <ul>
      <li>Microbiome-based machine learning models to predict childhood obesity trajectory</li>
    </ul>
  </li>
</ul> -->

  </nav>

  <article>
    <div class="project1">
      <img src="images/proj1_canva3.jpg" alt="vitamin D use and gut microbiome" width="100" height="70"> 
      <p>
        In this project, my colleagues and I developed a multiple linear model to investigate the relationship between vitamin D use, fecal metabolites, and microbiota, adjusting for covariables. The project used a cohort dataset of 575 infants. Vitamin D use and some prenatal variables were extracted from questionnaires; fecal metabolites and microbiota data were a readout of the NMR spectrum and high-throughput DNA sequencing, respectively.
      </p>
      

    </div>
    
  </article>

  <aside>
    <ul>
      <li>Web development Projects</li>
      <li>Microbiome Epidemiology Projects</li>
      <li>Machine Learning Projects</li>
      <li>Genomics Bioinformatics Projects</li>
    </ul>

  </aside>
</section>



  <!-- <h2 id="identifier" class="heading2">
    ðŸ¦ Microbiome Epidemiology <a name="topic1"></a>
  </h2>
  --- 
  
  <h3 id="identifier" class="heading3">
    Impact of vitamin D supplements on infant gut microbiome<a name="subtopic1"></a>
  </h3> 
  <br/>
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css">
  <div class="field is-grouped is-grouped-multiline">
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Linear Regression</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Logistic Regression</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Cohort study</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">R</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
    </div>
  
  <img src="images/proj1_canva3.jpg"/>
  
  <p id="identifier" class="p">
  In this project, my colleagues and I developed a multiple linear model to investigate the relationship between vitamin D use, fecal metabolites, and microbiota, adjusting for covariables. The project used a cohort dataset of 575 infants. Vitamin D use and some prenatal variables were extracted from questionnaires; fecal metabolites and microbiota data were a readout of the NMR spectrum and high-throughput DNA sequencing, respectively. 
  </p>
  <br/>
  [![](https://img.shields.io/badge/PDF-Open_research_poster-B44F99?logo=PDF)](pdf/Poster_vitD_microbiome_david.pdf) 
  [![](https://img.shields.io/badge/HTML-View%20artile-yellow)](https://www.mdpi.com/2218-273X/13/2/200)  -->
  
  ---
  <!-- <h3 id="identifier" class="heading3">
  Gut microbiota variations between IBD versus healthy adults<a name="subtopic2"></a> 
  </h3> 
  <br/>
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css">
  <div class="field is-grouped is-grouped-multiline">
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Mixed-Effect Model</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Data Transformation</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Multiple Test Correction</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">MaAsLin2</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
    </div>
  
  <img src="images/proj2_canva.png"/>
  
  <p id="identifier" class="p">
  To detect taxa differentially abundant in IBD patients, I built a generalized mixed-effect model with MaAsLin2 package, considering repeated samples as a random-effect variable. The input metagenomic data and metadata were derived from the openly available IBD dataset of the integrated human microbiome project (iHMP). 
  </p>
  <br/>
  [![](https://img.shields.io/badge/GitHub-View_on_GitHub-1182C3?logo=GitHub)](html/maaslin2-IBD-microbiome.html) 
  
  ---
  <h3 id="identifier" class="heading3">
  Benchmarking algorithms on zero-excessive metagenome data<a name="subtopic3"></a>
  </h3> 
  <br/>
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css">
  <div class="field is-grouped is-grouped-multiline">
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Zero-Inflation Model</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Zero-Hurdle Model</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Negative Binomial Model</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
   </div>
  
  <img src="images/proj3_canva.png"/>
  
  <p id="identifier" class="p">
  Four statistical algorithms, zero-inflated negative binomial, zero-inflated Poisson, zero-hurdle negative binomial, and zero-hurdle Poisson have been proposed in the literature to treat excessive zeroness (i.e., zero inflation) in various datatype, including microbiome count data. I benchmarked the four algorithms using microbiome dataset from Human Microbiome Project. Furthermore, I evaluated the performance based on AIC and likelihood ratio test.  
  </p>
  <br/>
  [![](https://img.shields.io/badge/GitHub-View_on_GitHub-0476D0?logo=GitHub)](html/zero-inflated-models-mcirobiome.html) 
  
  ---
  
  <h3 id="identifier" class="heading3">
  Differential analysis of microbiota data with over-dispersion using edgeR and DESeq2<a name="subtopic4"></a>
  </h3> 
  <br/>
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css">
  <div class="field is-grouped is-grouped-multiline">
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">edgeR</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">DESeq2</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  </div>
  
  <img src="images/proj4_canva2.png"/> 
  
  <p id="identifier" class="p">
  The microbiome is often over-dispersed and zero-inflated invalidating assumptions of conventional statistical models. The negative binomial algorithm can address the over-dispersion issue inherent in microbiome count data. In this project, I detected differentially-abundant taxa in IBD patients using the edgeR package that implements negative binomial modeling. 
  </p>
  <br/>
  
  [![](https://img.shields.io/badge/GitHub-View_on_GitHub-0476D0?logo=GitHub)](html/overdisp-zeroinflate-model.html) 
  
  ---
  
  <h3 id="identifier" class="heading3">
  Correlation network development for gut microbial taxa using SparCC<a name="subtopic5"></a> 
  </h3> 
  <br/>
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css">
  <div class="field is-grouped is-grouped-multiline">
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Network</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">SparCC</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  </div>
  
  <img src="images/sparcc_git_profile.jpg"/>
  
  <p id="identifier" class="p">
  Friedman and Alm developed SparCC to infer the correlation network between taxa from microbiome data, effectively dealing with compositionality. This project aims to demonstrate application of SparCC on a microbiome dataset, using Python and R language. 
  </p>
  <br/>
  
  [![](https://img.shields.io/badge/GitHub-View_on_GitHub-0476D0?logo=GitHub)](https://davidzhao1015.github.io/sparcc-relative-corr2/#1_install_package)
  
  ---
  <h3 id="identifier" class="heading3">
  Proportionality of gut microbiome to mitigrate spurious correlation<a name="subtopic6"></a> 
  </h3> 
  <br/>
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css">
  <div class="field is-grouped is-grouped-multiline">
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Relative Correlation</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">propr</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  </div>
  
  <img src="images/proj6_canva.png"/> 
  <p id="identifier" class="p">
  Lovell and co-authors (Lovell et al. 2015) proposed the proportionality method for relative correlation, as a valid alternative to absolute correlation. This project, thus, focused on illustrating the proportionality analysis of microbiome data using propr package by Quiin. Consistenet with previous findings, I observed proportionality is a precise indicator of absolute correlation, although sensitivity is limited. 
  </p>
  <br/>
  [![](https://img.shields.io/badge/GitHub-View_on_GitHub-0476D0?logo=GitHub)](https://davidzhao1015.github.io/association_analysis_compositional/)
  
  --- 
  <br/>
  <br/>
  <h2 id="identifier" class="heading2">
  ðŸ¤–Machine Learning<a name="topic2"></a> 
  </h2>
  --- 
  <h3 id="identifier" class="heading3">
  Microbiome-based machine learning models to predict childhood obesity trajectory<a name="subtopic7"></a> 
  </h3> 
  <br/>
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css">
  <div class="field is-grouped is-grouped-multiline">
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Predictive Modelling</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">Random Forest</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
    
     <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">XGBoost</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
    
     <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">caret</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  </div>
  
  <img src="images/fig3_proj1.png?raw=false"/>
  
  <p id="identifier" class="p">
  Knowing obesity risk earlier could help healthcare professionals manage children's weight more effectively. This project aims to use fecal microbiome data to predict future obesity risk with machine learning algorithms. Random Forest, XGBoost, GLMM, and Regularized Logistic classifiers were developed based on the data set of 2000+ infants. The optimal classifier achieved an AUC of 0.83 with a specificity of 73% and sensitivity of 85%. 
  </p>
  <br/>
  [![](https://img.shields.io/badge/PDF-Open_presentation_slides-C6466B?logo=PDF)](pdf/PPT_predict-child-obesity_davidzhao.pdf)
  [![](https://img.shields.io/badge/GitHub-View_on_GitHub-0476D0?logo=GitHub)](https://github.com/davidzhao1015/infant_microbiota_predict_obesity/blob/main/bmi_traj_pred_infant_microbiota.Rmd) 
  
  ---
  <br/>
  <br/>
  <h2 id="identifier" class="heading2">
  ðŸ§¬Genomic Bioinformatics<a name="topic3"></a> 
  </h2>
  --- 
  <h3 id="identifier" class="heading3"> 
  Comparative genomics of Lactobacillus reuteri isolates from gut and sourdough<a name="subtopic8"></a>  
  </h3> 
  <br/>
  
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bulma@0.9.4/css/bulma.min.css">
  <div class="field is-grouped is-grouped-multiline">
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">BLAST</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
  
    <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">MEGA 5</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
    
     <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">MAUVE</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
    
     <div class="control">
      <div class="tags has-addons">
        <a class="tag is-link is-small">KEGG</a>
        <a class="tag is-delete is-small"></a>
      </div>
    </div>
   </div>
  
  <img src="images/image_comparative-genomics.png?raw=false"/>
  
  <p id="identifier" class="p">
  In this project, our team analyzed the genomic adaption of strains from guts to sourdoughs. Using comparative genomics, we pinpointed particular genes under positive selection. In addition, I developed the in-house bioinformatic pipeline, detecting DNA variations among strains. 
  </p>
  <br/>
  [![](https://img.shields.io/badge/PDF-Open_presentation_slides-C6466B?logo=PDF)](pdf/PPT_phd_david.pdf)   -->
  
</body> 
  </html>